packages/@lwc/engine-dom/src/styles.ts

```diff
@@ -60,7 +60,8 @@ const stylesheetCache: Map<String, CacheData> = new Map();
 // Test utilities
 //
 
-if (process.env.NODE_ENV === 'development') {
+// @ts-ignore
+if (process.env.NODE_ENV !== 'production' && typeof __karma__ !== 'undefined') {
```

Comment:
We're already doing a similar check elsewhere: 

https://github.com/salesforce/lwc/blob/e9e5bc8a16fd54a3dd858ac6e646c46f1b45e5fe/packages/%40lwc/engine-dom/src/testFeatureFlag.ts#L10-L12